src/asymmetric_crypto/curve_25519/ed25519/certificate/mod.rs

```diff
@@ -53,6 +58,39 @@ impl Certificate {
     pub fn to_pem(&self) -> Result<String, CryptoCoreError> {
         Ok(self.inner.to_pem(LineEnding::LF)?)
     }
+
+    pub fn spki(&self) -> Result<Vec<u8>, CryptoCoreError> {
+        match &self.inner.tbs_certificate.extensions {
+            Some(exts) => {
+                let ski_list = exts
+                    .iter()
+                    .filter(|x| x.extn_id == ID_CE_SUBJECT_KEY_IDENTIFIER)
+                    .collect::<Vec<_>>();
+                if ski_list.is_empty() {
+                    return Err(CryptoCoreError::Certificate(format!(
+                        "Extension {ID_CE_SUBJECT_KEY_IDENTIFIER:?} not found"
+                    )));
+                }
+
+                if ski_list.len() > 1 {
+                    return Err(CryptoCoreError::Certificate(format!(
+                        "Multiple extensions {ID_CE_SUBJECT_KEY_IDENTIFIER:?} found"
+                    )));
+                }
+
+                let ski = ski_list
+                    .first()
+                    .ok_or(CryptoCoreError::Certificate(format!(
+                        "Cannot get first extension {ID_CE_SUBJECT_KEY_IDENTIFIER:?}"
+                    )))?;
+                let skid = SubjectKeyIdentifier::from_der(ski.extn_value.as_bytes())?;
+                Ok(skid.0.into_bytes())
```

Comment:
It seems you don't need ownership. You should be able to write this:
```suggestion
                let skid = SubjectKeyIdentifier::from_der(ski_list[0].extn_value.as_bytes())?;
                Ok(skid.0.into_bytes())
```